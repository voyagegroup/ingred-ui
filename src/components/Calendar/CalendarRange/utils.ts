import dayjs, { Dayjs } from "dayjs";

const isStart = (startDate: Dayjs | null, month: Dayjs, day: number) =>
  startDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isEnd = (endDate: Dayjs | null, month: Dayjs, day: number) =>
  endDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isBetween = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) =>
  (startDate &&
    endDate &&
    dayjs(new Date(month.year(), month.month(), day)).isAfter(
      startDate.format("YYYY-MM-DD"),
      "day",
    ) &&
    dayjs(new Date(month.year(), month.month(), day)).isBefore(
      endDate.format("YYYY-MM-DD"),
      "day",
    )) ??
  false;

export const getDayState = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) => {
  if (isStart(startDate, month, day)) {
    return "start";
  }
  if (isEnd(endDate, month, day)) {
    return "end";
  }
  if (isBetween(startDate, endDate, month, day)) {
    return "between";
  }
  return "none";
};
