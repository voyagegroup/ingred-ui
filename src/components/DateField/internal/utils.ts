export const getInputWidth = (input: string) =>
  [...input.split("")].reduce((acc, char) => {
    const code = char.charCodeAt(0);
    return acc + (code >= 0x3000 && code <= 0xff60 ? 2 : 1);
  }, 0);
