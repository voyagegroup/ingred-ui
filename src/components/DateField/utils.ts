import { useMemo } from "react";

type Sections = {
  start: number;
  end: number;
  value: string;
  editable: boolean;
};

/**
 * 何らかのフォーマットで入ってくる日付を開始位置と終了位置と値を持つセクションに分割する
 * useDateField で format された日付操作を汎用的に行うために必要なプロパティを返す
 * 例) 2023-01-02 -> [
 *   { start: 0, end: 3, value: "2023", editable: true },
 *   { start: 4, end: 4, value: "-", editable: false },
 *   { start: 5, end: 6, value: "01", editable: true },
 *   { start: 7, end: 7, value: "-", editable: false },
 *   { start: 8, end: 9, value: "02", editable: true }
 * ]
 *
 * @param formattedDate 何らかのフォーマットで入ってくる日付
 * @returns 開始位置と終了位置と値を持つセクション
 */
export const getSections = (formattedDate?: string | null) => {
  if (!formattedDate) {
    return [];
  }

  const sections: Sections[] = [];
  let start = 0;
  let isPrevCharDigit = !isNaN(Number(formattedDate[0]));

  for (let index = 1; index <= formattedDate.length; index++) {
    const currentChar = formattedDate[index];
    const isCurrentCharDigit =
      !isNaN(Number(currentChar)) && currentChar !== " ";

    if (
      isCurrentCharDigit !== isPrevCharDigit ||
      currentChar === " " ||
      index === formattedDate.length
    ) {
      sections.push({
        start,
        end: index - 1,
        value: formattedDate.slice(start, index),
        editable: isPrevCharDigit,
      });
      start = index;
      isPrevCharDigit = isCurrentCharDigit;
    }
  }

  return sections;
};

/**
 * 開始位置と終了位置と値を持つセクションをフォーマットされた日付に変換する
 */
export const formatString = (sectionsWithCharactor: Sections[]) =>
  sectionsWithCharactor.map((section) => section.value).join("");

type ReactRef<T> =
  | React.RefCallback<T>
  | React.MutableRefObject<T>
  | React.ForwardedRef<T>
  | string
  | null
  | undefined;

// from: https://github.com/voyagegroup/ingred-ui/blob/master/src/hooks/useMergeRefs.ts
export function useMergeRefs<T>(...refs: ReactRef<T>[]): React.Ref<T> {
  return useMemo(() => {
    if (refs.every((ref) => ref === null)) {
      return null;
    }
    return (refValue: T) => {
      for (const ref of refs) {
        if (typeof ref === "function") {
          ref(refValue);
        } else if (ref && typeof ref !== "string") {
          ref.current = refValue;
        }
      }
    };
  }, [refs]);
}
