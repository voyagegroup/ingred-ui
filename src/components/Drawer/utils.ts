/**
 * 相対値をpx値に変換するヘルパー関数
 * @param value - 変換する値（数値またはvw/vh/%/px等の単位付き文字列）
 * @param direction - Drawerの方向（変換時の基準値決定に使用）
 * @returns px値（数値）
 */
export const convertToPixels = (value: string | number, direction: "left" | "right" | "bottom"): number => {
  if (typeof value === "number") {
    return value; // 数値はpx扱い
  }
  
  // 文字列の場合は単位チェック
  const numericValue = parseFloat(value);
  
  if (value.includes("vw")) {
    return (numericValue / 100) * window.innerWidth;
  }
  if (value.includes("vh")) {
    return (numericValue / 100) * window.innerHeight;
  }
  if (value.includes("%")) {
    // %の場合は親要素基準だが、ここではviewport基準として扱う
    const isHorizontal = direction === "left" || direction === "right";
    return (numericValue / 100) * (isHorizontal ? window.innerWidth : window.innerHeight);
  }
  if (value.includes("px")) {
    return numericValue;
  }
  
  // 単位なしの文字列は警告を出す
  if (!isNaN(numericValue)) {
    console.warn(`Drawer: 文字列サイズには単位を指定してください。"${value}" → "${value}px" を推奨`);
    return numericValue; // 一応数値として扱う
  }
  
  // 無効な値の場合はエラー警告とフォールバック
  console.error(`Drawer: 無効なサイズ値です: "${value}". デフォルト値を使用します。`);
  return 400; // フォールバック
}; 