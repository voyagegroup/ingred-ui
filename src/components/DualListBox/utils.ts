import {
  CandidateItem,
  DualListBoxCandidateItem,
  DualListBoxItem,
} from "./DualListBox";

export const getCandidateItems = (
  candidateItems: DualListBoxCandidateItem[],
  selectedItems: DualListBoxItem[],
) => {
  const result: CandidateItem[] = [];
  candidateItems.forEach((candidateItem) => {
    if (candidateItem.items) {
      result.push({
        ...candidateItem,
        items: getCandidateItems(candidateItem.items, selectedItems),
      });
      return;
    }

    const targetItem = selectedItems.find(
      (selectedItem) => selectedItem.id === candidateItem.id,
    );
    if (targetItem) {
      result.push({
        ...candidateItem,
        selected: true,
      });
    } else {
      result.push(candidateItem);
    }
  });

  return result;
};
