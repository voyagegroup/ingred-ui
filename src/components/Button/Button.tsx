import * as React from "react";
import { fontSize } from "../Typography/Typography";

import { Props as BaseButtonProps } from "./internal/BaseButton";
import * as Styled from "./styled";
import { Theme, useTheme } from "../../themes";
import { hexToRgba } from "../../utils/hexToRgba";

export type ButtonSize = "small" | "medium" | "large";
export type ButtonColor = "primary" | "secondary" | "danger" | "clear";

export type ButtonColorStyle = {
  normal: {
    background: string;
    color: string;
    boxShadow: string;
    border: string;
  };
  hover: {
    background?: string;
    border: string;
  };
  active: {
    background?: string;
    boxShadow: string;
    border: string;
  };
};

export type Padding = {
  theme: Theme;
  size: ButtonSize;
  color: ButtonColor;
};

const getContainerColorStyles = (
  theme: Theme,
): { [P in ButtonColor]: ButtonColorStyle } => ({
  primary: {
    normal: {
      background: theme.palette.primary.main,
      color: theme.palette.text.white,
      boxShadow: `0px -2px ${hexToRgba(
        theme.palette.black,
        0.16,
      )} inset, 0px 2px ${hexToRgba(theme.palette.black, 0.08)}`,
      border: `1px solid ${theme.palette.primary.dark}`,
    },
    hover: {
      background: theme.palette.primary.dark,
      border: "none",
    },
    active: {
      background: theme.palette.primary.dark,
      boxShadow: `inset 0 2px ${hexToRgba(theme.palette.black, 0.16)}`,
      border: "none",
    },
  },
  secondary: {
    normal: {
      background: theme.palette.background.default,
      color: theme.palette.black,
      boxShadow: `0px -2px ${hexToRgba(
        theme.palette.black,
        0.16,
      )} inset, 0px 2px ${hexToRgba(theme.palette.black, 0.08)}`,
      border: `1px solid ${theme.palette.divider}`,
    },
    hover: {
      background: theme.palette.gray.highlight,
      border: `1px solid ${theme.palette.divider}`,
    },
    active: {
      background: theme.palette.gray.highlight,
      boxShadow: `inset 0 2px ${hexToRgba(theme.palette.black, 0.16)}`,
      border: `1px solid ${theme.palette.divider}`,
    },
  },
  danger: {
    normal: {
      background: theme.palette.danger.main,
      color: theme.palette.text.white,
      boxShadow: `0px -2px ${hexToRgba(
        theme.palette.black,
        0.16,
      )} inset, 0px 2px ${hexToRgba(theme.palette.black, 0.08)}`,
      border: `1px solid ${theme.palette.danger.dark}`,
    },
    hover: {
      background: theme.palette.danger.dark,
      border: "none",
    },
    active: {
      background: theme.palette.danger.dark,
      boxShadow: `inset 0 2px ${hexToRgba(theme.palette.black, 0.16)}`,
      border: "none",
    },
  },
  clear: {
    normal: {
      background: "none",
      color: theme.palette.gray.deepDark,
      boxShadow: "none",
      border: "none",
    },
    hover: {
      background: theme.palette.gray.light,
      border: "none",
    },
    active: {
      background: theme.palette.gray.main,
      boxShadow: "none",
      border: "none",
    },
  },
});

const paddingAtActive: Record<
  ButtonSize,
  { paddingTop: string; paddingBottom: string }
> = {
  small: {
    paddingTop: "7px",
    paddingBottom: "5px",
  },
  medium: {
    paddingTop: "11px",
    paddingBottom: "9px",
  },
  large: {
    paddingTop: "14px",
    paddingBottom: "12px",
  },
};

const verticalPadding: Record<ButtonSize, { padding: string }> = {
  small: {
    padding: "6px",
  },
  medium: {
    padding: "10px",
  },
  large: {
    padding: "13px",
  },
};

const getPadding = ({ theme, size, color }: Padding) => ({
  horizontalPadding: size === "small" ? `10px` : `${theme.spacing * 2}px`,
  verticalPadding: verticalPadding[size].padding,
  paddingTopAtActive: color === "clear" ? "" : paddingAtActive[size].paddingTop,
  paddingBottomAtActive:
    color === "clear" ? "" : paddingAtActive[size].paddingBottom,
});

export type ButtonProps = Omit<BaseButtonProps, "color"> & {
  /**
   * The component used for the root node.
   * Default: `<button />`
   */
  component?:
    | keyof JSX.IntrinsicElements
    | React.ComponentType<{ className: string }>;
  color?: ButtonColor;
  /**
   * Control whether "inline" or "block" Element.
   */
  inline?: boolean;
  size?: ButtonSize;
  onClick?: (event: React.MouseEvent<Element, MouseEvent>) => void;
  /**
   * If added this props, root node becomes `<a />`.
   */
  href?: string;
};

const Button = React.forwardRef<HTMLElement, ButtonProps>(
  (
    {
      component,
      children,
      color = "primary",
      inline = false,
      size = "medium",
      href,
      ...rest
    },
    ref,
  ) => {
    const theme = useTheme();
    const colorStyle = getContainerColorStyles(theme)[color];
    const {
      horizontalPadding,
      verticalPadding,
      paddingTopAtActive,
      paddingBottomAtActive,
    } = getPadding({
      theme,
      size,
      color,
    });

    const isLink = !!href;
    let anchorProps: any = {};
    if (isLink) {
      anchorProps = {
        as: component || "a",
        href,
      };
    }

    return (
      <Styled.ButtonContainer
        ref={ref}
        {...rest}
        as={component || "button"}
        {...anchorProps}
        inline={inline}
        verticalPadding={verticalPadding}
        horizontalPadding={horizontalPadding}
        paddingTopAtActive={paddingTopAtActive}
        paddingBottomAtActive={paddingBottomAtActive}
        normal={{ ...colorStyle.normal }}
        hover={{ ...colorStyle.hover }}
        active={{ ...colorStyle.active }}
        fontWeight={color === "secondary" ? "normal" : "bold"}
        fontSize={
          size === "small" ? `${fontSize["xs"]}px` : `${fontSize["md"]}px`
        }
      >
        {children}
      </Styled.ButtonContainer>
    );
  },
);

export default Button;
