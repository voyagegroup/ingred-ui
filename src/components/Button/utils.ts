import { Theme } from "../../themes";
import { ButtonColor, ButtonColorStyle } from "./Button";

export const getContainerColorStyles = (
  theme: Theme,
): { [P in ButtonColor]: ButtonColorStyle } => ({
  primary: {
    normal: {
      background: theme.palette.primary.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowPrimary"],
      border: `1px solid ${theme.palette.primary.dark}`,
    },
    hover: {
      background: theme.palette.primary.dark,
      border: "none",
    },
    active: {
      background: theme.palette.primary.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  basicLight: {
    normal: {
      background: theme.palette.background.default,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.divider}`,
    },
    hover: {
      background: theme.palette.gray.highlight,
      border: `1px solid ${theme.palette.divider}`,
    },
    active: {
      background: theme.palette.gray.highlight,
      boxShadow: theme.shadow["3dShadowActive"],
      border: `1px solid ${theme.palette.divider}`,
    },
  },
  basicDark: {
    normal: {
      background: theme.palette.basicDark.main,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.basicDark.deepDark}`,
    },
    hover: {
      background: theme.palette.basicDark.dark,
      border: "none",
    },
    active: {
      background: theme.palette.basicDark.dark,
      boxShadow: theme.shadow["3dShadowBasicActive"],
      border: "none",
    },
  },
  danger: {
    normal: {
      background: theme.palette.danger.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowDanger"],
      border: `1px solid ${theme.palette.danger.dark}`,
    },
    hover: {
      background: theme.palette.danger.dark,
      border: "none",
    },
    active: {
      background: theme.palette.danger.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  clear: {
    normal: {
      background: "none",
      color: theme.palette.black,
      boxShadow: "none",
      border: "none",
    },
    hover: {
      background: theme.palette.gray.light,
      border: "none",
    },
    active: {
      background: theme.palette.gray.main,
      boxShadow: "none",
      border: "none",
    },
  },
  primaryPale: {
    normal: {
      background: theme.palette.primaryPale.main,
      color: theme.palette.primaryPale.dark,
      boxShadow: theme.shadow["3dShadowPrimaryPale"],
      border: `1px solid ${theme.palette.primaryPale.light}`,
    },
    hover: {
      background: theme.palette.primaryPale.highlight,
      border: "none",
    },
    active: {
      background: theme.palette.primaryPale.highlight,
      boxShadow: theme.shadow["3dShadowPrimaryPaleActive"],
      border: "none",
    },
  },
});
