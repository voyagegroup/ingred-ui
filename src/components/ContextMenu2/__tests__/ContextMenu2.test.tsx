import React from "react";
import "@testing-library/jest-dom";
import { cleanup, fireEvent, act } from "@testing-library/react";
import { renderWithThemeProvider } from "../../../utils/renderWithThemeProvider";
import {
  ContextMenu2Container,
  ContextMenu2,
  ContextMenu2HeadingItem,
  ContextMenu2HelpTextItem,
  ContextMenu2TriggerItem,
  ContextMenu2ButtonItem,
  ContextMenu2CheckItem,
  ContextMenu2SwitchItem,
  ContextMenu2TextInputItem,
  ContextMenu2SeparatorItem,
  ContextMenu2ButtonControlsItem,
} from "../";
import Button from "../../Button";
import Icon from "../../Icon";

jest.mock("react-dom", () => {
  const original = jest.requireActual("react-dom");
  return {
    ...original,
    createPortal: (node: any) => node,
  };
});

// MEMO: ResizeObserver is not supported in jsdom
global.ResizeObserver = jest.fn().mockImplementation(() => ({
  observe: jest.fn(),
  unobserve: jest.fn(),
  disconnect: jest.fn(),
}));

describe("ContextMenu2 component testing", () => {
  afterEach(cleanup);

  test("ContextMenu2", async () => {
    const { asFragment, getByTestId } = renderWithThemeProvider(
      <ContextMenu2Container>
        <ContextMenu2
          trigger={
            <button type="button" data-testid="trigger">
              trigger
            </button>
          }
          width={316}
        >
          <ContextMenu2HeadingItem>ステータスを変更</ContextMenu2HeadingItem>
          <ContextMenu2HelpTextItem prepend={<Icon name="question" />}>
            ヘルプテキスト
          </ContextMenu2HelpTextItem>
          <ContextMenu2SeparatorItem />
          <ContextMenu2ButtonItem
            prepend={<Icon name="import" />}
            onClick={() => {}}
          >
            アイコンボタン
          </ContextMenu2ButtonItem>
          <ContextMenu2ButtonItem onClick={() => {}}>
            ボタン
          </ContextMenu2ButtonItem>
          <ContextMenu2ButtonItem color="danger" onClick={() => {}}>
            赤ボタン
          </ContextMenu2ButtonItem>
          <ContextMenu2
            trigger={
              <ContextMenu2TriggerItem append={100}>
                入れ子トリガー
              </ContextMenu2TriggerItem>
            }
          >
            <ContextMenu2ButtonItem onClick={() => {}}>
              入れ子ボタン
            </ContextMenu2ButtonItem>
          </ContextMenu2>
          <ContextMenu2CheckItem
            checked={true}
            prepend={<Icon name="image" />}
            onChange={() => {}}
          >
            含む
          </ContextMenu2CheckItem>
          <ContextMenu2CheckItem
            checked={false}
            prepend={<Icon name="image" />}
            onChange={() => {}}
          >
            含まない
          </ContextMenu2CheckItem>
          <ContextMenu2CheckItem checked={false} onChange={() => {}}>
            アイコンなしチェック1
          </ContextMenu2CheckItem>
          <ContextMenu2CheckItem checked={false} onChange={() => {}}>
            アイコンなしチェック2
          </ContextMenu2CheckItem>
          <ContextMenu2SwitchItem checked={false} onChange={() => {}}>
            スイッチ
          </ContextMenu2SwitchItem>
          <ContextMenu2TextInputItem
            value="入力内容"
            placeholder="プレイスホルダー"
            onChange={() => {}}
          />
          <ContextMenu2ButtonControlsItem>
            <Button size="small" color="clear">
              キャンセル
            </Button>
            <Button size="small">適用</Button>
          </ContextMenu2ButtonControlsItem>
          <ContextMenu2ButtonItem onClick={() => alert("最後の項目")}>
            最後の項目
          </ContextMenu2ButtonItem>
        </ContextMenu2>
      </ContextMenu2Container>,
    );
    await act(async () => {
      fireEvent.click(getByTestId("trigger"));
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
