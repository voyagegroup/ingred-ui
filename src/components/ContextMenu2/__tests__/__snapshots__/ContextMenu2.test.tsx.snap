// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextMenu2 component testing ContextMenu2 1`] = `
<DocumentFragment>
  <button
    aria-controls=":r1:"
    aria-expanded="true"
    aria-haspopup="dialog"
    data-floating-ui-inert=""
    data-testid="trigger"
    type="button"
  >
    trigger
  </button>
  <span
    aria-hidden="true"
    data-floating-ui-inert=""
    style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
    tabindex="-1"
  />
  <span
    data-floating-ui-focus-guard=""
    data-floating-ui-inert=""
    data-type="outside"
    role="button"
    style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
    tabindex="0"
  />
  <span
    aria-owns=":r3:"
    data-floating-ui-inert=""
    style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
  />
  <span
    data-floating-ui-focus-guard=""
    data-floating-ui-inert=""
    data-type="inside"
    role="button"
    style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
    tabindex="0"
  />
  <div
    aria-orientation="vertical"
    class="sc-dHKmnV bWxvwo"
    data-floating-ui-focusable=""
    id=":r1:"
    role="dialog"
    style="position: absolute; left: 0px; top: 0px; width: 316px; min-width: auto; transform: translate(0px, 5px); max-width: 0px; max-height: 0px;"
    tabindex="-1"
  >
    <div
      class="sc-hLyRwt jRdXFI"
    >
      <p
        class="sc-iLBnws fVxPFs"
      >
        ステータスを変更
      </p>
      <div
        class="sc-fGlNzy hmnNFu"
      >
        <div>
          <span
            class="sc-fmZSGO naXpd"
            size="18"
          >
            <svg
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M12,22A10,10,0,1,1,22,12,10,10,0,0,1,12,22Zm0-2a8,8,0,1,0-8-8A8,8,0,0,0,12,20Zm-1-5h2v2H11Zm2-1.645V14H11V12.5a1,1,0,0,1,1-1,1.5,1.5,0,1,0-1.471-1.794L8.567,9.313A3.5,3.5,0,1,1,13,13.355Z"
                fill="#596978"
              />
            </svg>
          </span>
        </div>
        <p
          class="sc-eGjrzz eiKhWT"
        >
          ヘルプテキスト
        </p>
      </div>
      <hr
        class="sc-hawdNJ bFRDZp"
      />
      <button
        class="sc-hKDTPf jOFScb"
        tabindex="-1"
        type="button"
      >
        <span
          class="sc-bXdtCk cPSCTd"
        >
          <span
            class="sc-fmZSGO naXpd"
            size="18"
          >
            <svg
              viewBox="0 0 18 18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h18v18H0z"
                fill="none"
              />
              <path
                d="M10.25,8.25H14l-4.5,4.5L5,8.25H8.75V3h1.5ZM3.5,15h12V9.75H17v6a.75.75,0,0,1-.75.75H2.75A.75.75,0,0,1,2,15.75v-6H3.5Z"
                fill="#596978"
                transform="translate(-0.5 -0.75)"
              />
            </svg>
          </span>
        </span>
        アイコンボタン
      </button>
      <button
        class="sc-hKDTPf jOFScb"
        tabindex="-1"
        type="button"
      >
        ボタン
      </button>
      <button
        class="sc-hKDTPf xhtyq"
        color="danger"
        tabindex="-1"
        type="button"
      >
        赤ボタン
      </button>
      <button
        aria-expanded="false"
        aria-haspopup="dialog"
        class="sc-llIIlC fPDZgD"
        type="button"
      >
        <span
          class="sc-kThouk krVvIl"
        >
          入れ子トリガー
        </span>
        <span
          class="sc-dTdQuR ddabir"
        >
          100
        </span>
        <span
          class="sc-fmZSGO naXpd"
          size="18"
        >
          <svg
            viewBox="0 0 20 20"
          >
            <path
              d="M10.9 10 6.8 5.9l1.1-1.2 5.3 5.3L8 15.3l-1.1-1.2z"
              fill="#596978"
            />
          </svg>
        </span>
      </button>
      <button
        class="sc-jMpmlX eRtVlL"
        tabindex="-1"
        type="button"
      >
        <span
          class="sc-gwsNht bjkGLQ"
        >
          <span
            class="sc-fmZSGO naXpd"
            size="18"
          >
            <svg
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.828 21L4.808 21.02L4.787 21H2.992C2.72881 20.9997 2.4765 20.895 2.29049 20.7088C2.10448 20.5226 2 20.2702 2 20.007V3.993C2.00183 3.73038 2.1069 3.47902 2.29251 3.29322C2.47813 3.10742 2.72938 3.00209 2.992 3H21.008C21.556 3 22 3.445 22 3.993V20.007C21.9982 20.2696 21.8931 20.521 21.7075 20.7068C21.5219 20.8926 21.2706 20.9979 21.008 21H4.828ZM20 15V5H4V19L14 9L20 15ZM20 17.828L14 11.828L6.828 19H20V17.828ZM8 11C7.46957 11 6.96086 10.7893 6.58579 10.4142C6.21071 10.0391 6 9.53043 6 9C6 8.46957 6.21071 7.96086 6.58579 7.58579C6.96086 7.21071 7.46957 7 8 7C8.53043 7 9.03914 7.21071 9.41421 7.58579C9.78929 7.96086 10 8.46957 10 9C10 9.53043 9.78929 10.0391 9.41421 10.4142C9.03914 10.7893 8.53043 11 8 11Z"
                fill="#596978"
              />
            </svg>
          </span>
        </span>
        含む
        <span
          class="sc-dYOLZc evqSwF"
        >
          <span
            class="sc-fmZSGO naXpd"
            size="18"
          >
            <svg
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.99999 15.172L19.192 5.979L20.607 7.393L9.99999 18L3.63599 11.636L5.04999 10.222L9.99999 15.172Z"
                fill="#0B82F4"
              />
            </svg>
          </span>
        </span>
      </button>
      <button
        class="sc-jMpmlX hhhSDo"
        tabindex="-1"
        type="button"
      >
        <span
          class="sc-gwsNht bjkGLQ"
        >
          <span
            class="sc-fmZSGO naXpd"
            size="18"
          >
            <svg
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4.828 21L4.808 21.02L4.787 21H2.992C2.72881 20.9997 2.4765 20.895 2.29049 20.7088C2.10448 20.5226 2 20.2702 2 20.007V3.993C2.00183 3.73038 2.1069 3.47902 2.29251 3.29322C2.47813 3.10742 2.72938 3.00209 2.992 3H21.008C21.556 3 22 3.445 22 3.993V20.007C21.9982 20.2696 21.8931 20.521 21.7075 20.7068C21.5219 20.8926 21.2706 20.9979 21.008 21H4.828ZM20 15V5H4V19L14 9L20 15ZM20 17.828L14 11.828L6.828 19H20V17.828ZM8 11C7.46957 11 6.96086 10.7893 6.58579 10.4142C6.21071 10.0391 6 9.53043 6 9C6 8.46957 6.21071 7.96086 6.58579 7.58579C6.96086 7.21071 7.46957 7 8 7C8.53043 7 9.03914 7.21071 9.41421 7.58579C9.78929 7.96086 10 8.46957 10 9C10 9.53043 9.78929 10.0391 9.41421 10.4142C9.03914 10.7893 8.53043 11 8 11Z"
                fill="#596978"
              />
            </svg>
          </span>
        </span>
        含まない
        <span
          class="sc-dYOLZc evqSwF"
        />
      </button>
      <button
        class="sc-jMpmlX hhhSDo"
        tabindex="-1"
        type="button"
      >
        アイコンなしチェック1
        <span
          class="sc-dYOLZc evqSwF"
        />
      </button>
      <button
        class="sc-jMpmlX hhhSDo"
        tabindex="-1"
        type="button"
      >
        アイコンなしチェック2
        <span
          class="sc-dYOLZc evqSwF"
        />
      </button>
      <button
        class="sc-hBDmJg ljnXvO"
        tabindex="-1"
        type="button"
      >
        スイッチ
        <span
          class="sc-hoLldG cuapxF"
        >
          <span
            class="sc-jZhnRx cAakXI"
          />
        </span>
      </button>
      <div
        class="sc-dACwDz etXABJ"
      >
        <input
          data-ios="false"
          placeholder="プレイスホルダー"
          tabindex="-1"
          value="入力内容"
        />
      </div>
      <ul
        class="sc-iayzNI livxWh"
      >
        <li>
          <button
            active="[object Object]"
            borderradius="4px"
            class="sc-gfUXbm sc-cTIdZS bZssAy cWFzfe"
            color="clear"
            font-size="12px"
            height="28px"
            hover="[object Object]"
            normal="[object Object]"
            paddinginline="6px"
          >
            キャンセル
          </button>
        </li>
        <li>
          <button
            active="[object Object]"
            borderradius="4px"
            class="sc-gfUXbm sc-cTIdZS bZssAy ghgdFh"
            color="primary"
            font-size="12px"
            height="28px"
            hover="[object Object]"
            normal="[object Object]"
            paddinginline="6px"
          >
            適用
          </button>
        </li>
      </ul>
      <button
        class="sc-hKDTPf jOFScb"
        tabindex="-1"
        type="button"
      >
        最後の項目
      </button>
    </div>
  </div>
  <span
    data-floating-ui-focus-guard=""
    data-floating-ui-inert=""
    data-type="inside"
    role="button"
    style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
    tabindex="0"
  />
  <span
    data-floating-ui-focus-guard=""
    data-floating-ui-inert=""
    data-type="outside"
    role="button"
    style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: fixed; white-space: nowrap; width: 1px; top: 0px; left: 0px;"
    tabindex="0"
  />
</DocumentFragment>
`;
